agent/collector.go

```diff
@@ -270,12 +271,23 @@ func (l *Collector) postToAPI(endpoint config.APIEndpoint, checkPath string, bod
 	req.Header.Add("X-Dd-Hostname", l.cfg.HostName)
 	req.Header.Add("X-Dd-Processagentversion", Version)
 
+	// we create a context with timeout and attach it to the http request, to make sure that it doesn't get stuck on
+	// any circumstances
+	start := time.Now()
+	ctx, cancel := context.WithTimeout(context.Background(), HTTPTimeout)
+	defer cancel()
+	req.WithContext(ctx)
+
 	resp, err := l.httpClient.Do(req)
 	if err != nil {
 		if isHTTPTimeout(err) {
 			responses <- errResponse("Timeout detected on %s, %s", url, err)
 		} else {
-			responses <- errResponse("Error submitting payload to %s: %s", url, err)
+			if time.Since(start) >= HTTPTimeout {
```

Comment:
I'd rather just leave the error handling as before for simplicity. it will say timeout either way (from my testing the error message with a context cancel was: `req err: Get https://google.com: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers`

Reply:
Really? If I set context timeout shorter than client timeout I got `Get http://127.0.0.1:3001: context deadline exceeded`.

Reply:
But yea, we could just keep the original checks.